### PR TITLE
Bug 1191023 - Fix failure in test_settings_do_not_track.py

### PR DESCRIPTION
**Pull request by mwargers**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9cdc9ad8c3f262b5dbed32c16277a63fb544a23f

File: tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/do_not_track.py [hunk 1]
```diff
@@ -9,27 +9,33 @@
...
-            *self._allow_tracking_checkbox_locator)
-        label.tap()
-        radio = label.find_element(By.TAG_NAME, 'input')
-        Wait(self.marionette).until(expected.element_selected(radio))
+        element = self.marionette.find_element(
+            *self._allow_tracking_radio_locator)
+        element.tap()
+        # The following should work, but doesn't, see bug 1113742. We use execute_script instead, for now
+        # Wait(self.marionette).until(expected.element_selected(element))
+        Wait(self.marionette).until(lambda m:
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/do_not_track.py]: We use the workaround also in Keyboard and Settings. I'd suggest to factorize it in `Base.py`, like:
```python
 # TODO: Remove me once bug 1113742 is fixed
def wait_on_check_state_for_shadow_dom_element(self, element, checked=True):
    Wait(self.marionette).until(lambda m:
            m.execute_script("return arguments[0].wrappedJSObject.checked", [element]) is checked)
```

and then here:
```python
element.tap()
self.wait_on_check_state_for_shadow_dom_element(element)
```

File: tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/do_not_track.py [hunk 2]
```diff
@@ -9,27 +9,33 @@
 
 class DoNotTrack(Base):
 
-    _allow_tracking_checkbox_locator = (By.XPATH, '//li/label[span[@data-l10n-id="allowTracking"]]')
-    _disallow_tracking_checkbox_locator = (By.XPATH, '//li/label[span[@data-l10n-id="doNotTrackActions"]]')
-    _do_not_have_pref_on_tracking_checkbox_locator = (By.XPATH, '//li/label[span[@data-l10n-id="doNotHavePrefOnTracking"]]')
+    _allow_tracking_radio_locator = (By.CSS_SELECTOR, 'gaia-radio[name="privacy.donottrackheader.value"][value="0"]')
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/do_not_track.py]: That's :+1:  that we finally get rid of these XPATH locators :)

File: tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/do_not_track.py (hunk 1 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/do_not_track.py]: I don't see that much benefit in doing this, but I'll do it.
I can't use this here at least: http://mxr.mozilla.org/gaia/source/tests/python/gaia-ui-tests/gaiatest/apps/contacts/regions/settings_form.py#43

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=890fde0f95eeff237a8d555cdca2fbde4d49158d

**Pull request merged**